schema/messaging.proto

```diff
@@ -0,0 +1,508 @@
+};
+
+// This resource is defined in google/showcase/v1alpha3/user.proto. Since
+// google/showcase/v1alpha3/user.proto is not imported from this file, the
+// resource needs to be defined here to be used in this API.
+option (google.api.resource_definition) = {
+  name: "User",
+  path: "users/{user_id}"
+};
+
+// A resource set defining the possible parent paths of a
+// message.
+option (google.api.resource_set_definition) = {
+  name: "MessageParent"
+  resources: [{
+    name: "Profile",
+    path: "users/{user_id}/profile"
+  }]
+  resource_references: [
+    "google.showcase.v1alpha3.Room"
+  ]
+};
+
+// A simple messaging service that implements chat rooms and profile posts.
+//
+// This messaging service showcases the features that API clients
+// generated by gapic-generators implement.
+service Messaging {
+  // Creates a room. The fields "room.name", "room.create_time", and
+  // "room.update_time" will be ignored if passed to this method.
+  rpc CreateRoom(CreateRoomRequest) returns (Room) {
+    option (google.api.http) = {
+      post: "/v1alpha3/rooms"
+      body: "*"
+    };
+
+    // This option signifies that the resulting generated client could include
+    // a method overload which accepts the following listed fields as
+    // parameters.
+    option (google.api.method_signature) = {
+      fields: ["room.display_name", "room.description"]
+    };
+  }
+
+  // Retrieves the Room with the given uri.
+  rpc GetRoom(GetRoomRequest) returns (Room) {
+    option (google.api.http) = {
+      get: "/v1alpha3/{name=rooms/*}"
+    };
+
+    // This option signifies that the resulting generated client could include
+    // a method overload which accepts the following listed fields as
+    // parameters.
+    option (google.api.method_signature) = {
+      fields: ["name"]
+    };
+  }
+
+  // Updates a room.
+  rpc UpdateRoom(UpdateRoomRequest) returns (Room) {
+    option (google.api.http) = {
+      patch: "/v1alpha3/{room.name=rooms/*}"
+      body: "*"
+    };
+  }
+
+  // Deletes a room and all of it's messages.
+  rpc DeleteRoom(DeleteRoomRequest) returns (google.protobuf.Empty) {
+    option (google.api.http) = {
+      delete: "/v1alpha3/{name=rooms/*}"
+    };
+
+    // This option signifies that the resulting generated client could include
+    // a method overload which accepts the following listed fields as
+    // parameters.
+    option (google.api.method_signature) = {
+      fields: ["name"]
+    };
+  }
+
+  // Lists all chat rooms.
+  rpc ListRooms(ListRoomsRequest) returns (ListRoomsResponse) {
+    option (google.api.http) = {
+      get: "/v1alpha3/rooms"
+    };
+  }
+
+  // Creates a message. If the parent field corresponds to a RoomMessage
+  // resource, the message is understood to be a message in a chatroom. If the
+  // parent field is a ProfileMessage resource name, the message is understood
+  // to be a post on a user profile. The fields "message.name",
```

Comment:
Omit this last sentence (as discussed above).

Reply:
Done.